src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java

```diff
@@ -45,54 +45,158 @@
+            feedbackQuestionAttributes.questionMetaData = questionMetaData;
+            return this;
+        }
+
+        public Builder withQuestionDescription(Text questionDescription) {
+            feedbackQuestionAttributes.setQuestionDescription(questionDescription);
+            return this;
+        }
+
+        public Builder withQuestionNumber(int questionNumber) {
+            feedbackQuestionAttributes.questionNumber = questionNumber;
+            return this;
+        }
+
+        public Builder withQuestionType(FeedbackQuestionType questionType) {
+            feedbackQuestionAttributes.questionType = questionType;
+            return this;
+        }
+
+        public Builder withGiverType(FeedbackParticipantType giverType) {
+            feedbackQuestionAttributes.giverType = giverType;
+            return this;
+        }
+
+        public Builder withRecipientType(FeedbackParticipantType recipientType) {
+            feedbackQuestionAttributes.recipientType = recipientType;
+            return this;
+        }
+
+        public Builder withNumOfEntitiesToGiveFeedbackTo(int numOfEntitiesToGiveFeedbackTo) {
+            feedbackQuestionAttributes.numberOfEntitiesToGiveFeedbackTo = numOfEntitiesToGiveFeedbackTo;
+            return this;
+        }
+
+        public Builder withShowResponseTo(List<FeedbackParticipantType> showResponseTo) {
+            feedbackQuestionAttributes.showResponsesTo = new ArrayList<>(showResponseTo);
+            return this;
+        }
+
+        public Builder withShowGiverNameTo(List<FeedbackParticipantType> showGiverNameTo) {
+            feedbackQuestionAttributes.showGiverNameTo = new ArrayList<>(showGiverNameTo);
+            return this;
+        }
+
+        public Builder withShowRecipientNameTo(List<FeedbackParticipantType> showRecipientNameTo) {
+            feedbackQuestionAttributes.showRecipientNameTo = new ArrayList<>(showRecipientNameTo);
+            return this;
+        }
+
+        public Builder withCreatedAt(Instant createdAt) {
+            feedbackQuestionAttributes.createdAt = createdAt;
+            return this;
+        }
+
+        public Builder withUpdatedAt(Instant updatedAt) {
+            feedbackQuestionAttributes.updatedAt = updatedAt;
+            return this;
+        }
+
+        public Builder withFeedbackQuestionId(String feedbackQuestionId) {
+            feedbackQuestionAttributes.feedbackQuestionId = feedbackQuestionId;
+            return this;
+        }
+
+        public FeedbackQuestionAttributes build() {
+            feedbackQuestionAttributes.questionDescription =
+                    SanitizationHelper.sanitizeForRichText(feedbackQuestionAttributes.questionDescription);
+            feedbackQuestionAttributes.removeIrrelevantVisibilityOptions();
+            return feedbackQuestionAttributes;
+        }
+
+        public FeedbackQuestionAttributes buildWithoutRemovingIrrelevantVisibilityOptions() {
```

Comment:
Can't quite figure out why this method is necessary; can you explain?

Reply:
> But in some of the tests, they use the empty constructor and assign the attributes line by line, without making a call to this method. So this method is to replicate that process.

To me, this looks like the tests are not correctly written. If we are no testing what will happend (using `removeIrrelevantVisibilityOptions`) in tests, what's the point of having these tests?